universal-application-tool-0.0.1/app/views/questiontypes/ApplicantQuestionRendererImpl.java

```diff
@@ -0,0 +1,78 @@
+package views.questiontypes;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static j2html.TagCreator.div;
+
+import com.google.common.collect.ImmutableSet;
+import j2html.tags.ContainerTag;
+import j2html.tags.Tag;
+import play.i18n.Messages;
+import services.MessageKey;
+import services.applicant.ValidationErrorMessage;
+import services.applicant.question.ApplicantQuestion;
+import views.BaseHtmlView;
+import views.components.TextFormatter;
+import views.style.ApplicantStyles;
+import views.style.ReferenceClasses;
+import views.style.Styles;
+
+/**
+ * Superclass for all applicant question renderers with input field(s) for the applicant to answer
+ * the question.
+ */
+public abstract class ApplicantQuestionRendererImpl implements ApplicantQuestionRenderer {
```

Comment:
suggestion (non-blocking): looks like this and the c'tor could be package-private rather than `public`?

Reply:
Done!